tests/unit/components/input-with-icon/InputWithIcon.spec.jsx

```diff
@@ -0,0 +1,29 @@
+import { render, screen } from '@testing-library/react';
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon';
+
+describe('InputWithIcon component', () => {
+    const props = {
+        value: 'value',
+        onClear: () => { },
+        startIcon: <div data-testid='startIcon' />
+    }
+    beforeEach(() => {
+        render(<InputWithIcon {...props} />)
+    })
+    it('renders container', () => {
+        const inputWithIconContainer = screen.getByTestId('inputWithIconContainer')
+        expect(inputWithIconContainer).toBeInTheDocument()
+    })
+    it('renders start icon that is passed in props', () => {
+        const startIcon = screen.getByTestId('startIcon')
+        expect(startIcon).toBeInTheDocument()
+    })
+    it('renders inputBase', () => {
+        const inputWithIconInputBase = screen.getByTestId('inputWithIconInputBase')
+        expect(inputWithIconInputBase).toBeInTheDocument()
+    })
+    it('renders clear icon when input value is not empty', () => {
+        const clearIcon = screen.getByTestId('clearIcon')
+        expect(clearIcon).toBeInTheDocument()
+    });
```

Comment:
Let`s add additional test when value is empty.

Reply:
Done, thanks